src/app/api/auth/lib/nextauth-options.ts

```diff
@@ -22,6 +23,67 @@ export const verifySession = async (token: string | undefined) => {
   }
 }
 
+/**
+ * Refresh access token
+ * @param clientId Client ID
+ * @param clientSecret Client Secret
+ * @param refreshToken Refresh token
+ * @returns object — New access token and refresh token
+ */
+export const refreshAccessToken = async (
+  token: JWT,
+  clientId: string,
+  clientSecret: string,
+  refreshToken: string,
+) => {
+  try {
+    authLogger.debug('Refreshing access token')
+
+    const url =
+      'https://github.com/login/oauth/access_token?' +
+      new URLSearchParams({
+        client_id: clientId,
+        client_secret: clientSecret,
+        refresh_token: refreshToken,
+        grant_type: 'refresh_token',
+      })
+
+    const response = await fetch(url, {
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded',
+      },
+      method: 'POST',
+    }).then(async (res) => {
+      const responseText = await res.text()
+      const entries = new URLSearchParams(responseText)
+      const params = Object.fromEntries(entries)
+
+      if (params.error) {
+        throw new Error(params.error_description)
+      }
+
+      return params
+    })
+
+    return {
+      accessToken: response.access_token,
+      // Access token expiration is provided as number in seconds until expiration (value is always 8 hours)
+      accessTokenExpires: Date.now() + Number(response.expires_in) * 1000,
+      refreshToken: response.refresh_token,
+      // Refresh token expiration is provided as number of seconds until expiration (value is always 6 months)
+      refreshTokenExpires:
+        Date.now() + Number(response.refresh_token_expires_in) * 1000,
+    }
```

Comment:
multiplying by 1000 because of milliseconds to seconds conversion? Could define a `const` for greater readability.
```suggestion

    const milliseconds_to_seconds = 1000
    return {
      accessToken: response.access_token,
      // Access token expiration is provided as number in seconds until expiration (value is always 8 hours)
      accessTokenExpires: Date.now() + Number(response.expires_in) * milliseconds_to_seconds,
      refreshToken: response.refresh_token,
      // Refresh token expiration is provided as number of seconds until expiration (value is always 6 months)
      refreshTokenExpires:
        Date.now() + Number(response.refresh_token_expires_in) * milliseconds_to_seconds,
    }
```